components/content-service/pkg/archive/tar_test.go

```diff
@@ -94,3 +95,65 @@ func TestBuildTarbalMaxSize(t *testing.T) {
 		os.RemoveAll(c)
 	}
 }
+
+func TestExtractTarbal(t *testing.T) {
+	files := []struct {
+		Name        string
+		ContentSize int64
+		Uid         int
+	}{
+		// would be nice to test this with different uids and even uid mappings, but it would reuqire to run the test privileged
```

Comment:
You could do something like the initializer tests used to do: run all tests you can and skip the ones you can't because you don't have the privileges. 

Also, it's quite uncommon in our Go unit tests to NOT have a table driven test. Is this really the only case you want to test? I'm thinking about symlinks, empy tar files, broken tar files, hardlinks, very long filenames, broken mappings.

Reply:
> You could do something like the initializer tests used to do: run all tests you can and skip the ones you can't because you don't have the privileges.

I cannot test any uid mapping or can I?

Reply:
Per our offline chat, I made the test array-based to align with other tests and added a case for an empty tar.